File: Swift/프로그래머스/0/120852. 소인수분해/소인수분해.swift
import Foundation

func solution(_ n:Int) -> [Int] {
    var factors = [Int]()
    
    var num = n
    
    var i = 2
    while i * i <= num {
        if num % i == 0 {
            factors.append(i)
            
            while num % i == 0 {
                num = num / i
            }
        } else {
            i += 1
        }
    }
    
    if num > 1 {
        factors.append(num)
    }
    
    return factors
}

File: Python3/프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    factors = []
    
    i = 2
    while i * i <= n:
        if n % i == 0:
            factors.append(i)
            while n % i == 0:
                n //= i
        else:
            i += 1
    
    if n > 1:
        factors.append(n)
    
    return factors